guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,750 @@
+We can now move on to model creation and training.
+
+### Model creation
+
+We'll use the KerasCV API to construct a RetinaNet model.
+In this tutorial we use a pretrained ResNet50 backbone, initializing the
+weights to weights produced by training a classification model on the imagenet
+dataset.
+
+KerasCV makes it easy to construct a `RetinaNet` with any of the KerasCV
+backbones.  Simply use one of the presets for the architecture you'd like!
+
+For example:
+"""
+
+model = keras_cv.models.RetinaNet.from_preset(
+    "resnet50_v2_imagenet",
+    # number of classes to be used in box classification
+    num_classes=len(class_mapping) + 1,
+    # For more info on supported bounding box formats, visit
+    # https://keras.io/api/keras_cv/bounding_box/
+    bounding_box_format="xywh",
+)
+
+"""
+That is all it takes to construct a KerasCV RetinaNet.  The RetinaNet accepts
+tuples of dense image Tensors and bounding box dictionaries to `fit()` and
+`train_on_batch()`
+
+This matches what we have constructed in our input pipeline above.
+"""
+
+"""
+### Optimizer
+
+Next, let's construct a learning rate schedule and optimizer for our model.
+"""
+base_lr = 0.005 * (BATCH_SIZE / 16)
+lr_decay = tf.keras.optimizers.schedules.PiecewiseConstantDecay(
+    boundaries=[12000 * 16, 16000 * 16],
+    values=[base_lr, 0.1 * base_lr, 0.01 * base_lr],
+)
+
+# including a global_clipnorm is extremely important in object detection tasks
+optimizer = tf.keras.optimizers.SGD(
+    learning_rate=lr_decay, momentum=0.9, global_clipnorm=10.0
+)
+
+"""
+**Note: be sure to freeze BatchNormalization layers***
+
+And important but easy to miss step in training your own object detection model
+is freezing the `BatchNormalization` layers in your backbone.
+We felt it was a violation of user expectations to automatically do this,
+so instead we recommend you do it in your training loops.
+"""
+
+for layer in model.backbone.layers:
+    if isinstance(layer, keras.layers.BatchNormalization):
+        layer.trainable = False
+
+"""
+### Metric Evaluation
+
+Just like any other metric, you can pass the `KerasCV` object detection metrics
+to `compile()`.  The most popular Object Detection metrics are COCO metrics,
+which were published alongside the MSCOCO dataset.  KerasCV provides an easy
+to use suite of COCO metrics. under the `keras_cv.metrics.BoxCOCOMetrics`
+symbol:
+"""
+coco_metrics = keras_cv.metrics.BoxCOCOMetrics(
+    bounding_box_format="xywh", evaluate_freq=128
+)
+"""
+Due to the high computational cost of computing COCO metrics, the KerasCV
+`BoxCOCOMetrics` component requires an `evaluate_freq` parameter to be passed to
+its constructor.  Every `evaluate_freq`-th call to `update_state()`, the metric
+will recompute the result.  In between invocations, a cached version of the
+result will be returned.
+
+To force an evaluation, you may call `coco_metrics.result(force=True)`:
+"""
+model.compile(
+    classification_loss="focal",
```

Comment:
These loss functions deserve some (even brief) explanation since they are custom to this problem.